docs/SpaceTemplates.rst

```diff
@@ -0,0 +1,229 @@
+       Steps to create a new space template.
+
+    -  :ref:`Extending existing templates <PLFDevGuide.SpaceTemplates.Extend>`
+       Steps for extending existing space templates. Only applications list can be extended.
+
+    -  :ref:`Defining an application handler for template <PLFDevGuide.SpaceTemplates.handler.CreateNew>`
+       Steps to define an application handler for a space template.
+
+.. tip:: In order to extend existing space templates or create a new one, you must have your own :ref:`extension <PLFDevGuide.eXoAdd-ons.PortalExtension>`.
+This section assumes that you have already a working extension where you can plug the configuration.
+
+.. tip:: All the configuration samples given in this chapter MUST be added to the portal configuration file of the extension war: *custom-extension.war!/WEB-INF/conf/configuration.xml*.
+
+
+.. _PLFDevGuide.SpaceTemplates.CreateNew:
+
+============================
+Creating a new template
+============================
+
+You can define a new space template by configuration using ``registerSpaceTemplatePlugin`` method of the ``org.exoplatform.social.core.space.spi.SpaceTemplateService``.
+A space template consists of *name*, *visibility*, *registration*, *banner*, *home application* and *applications list* fields.
+To add a space template, add the following configuration :
+
+    .. code:: xml
+
+        <external-component-plugins>
+            <target-component>org.exoplatform.social.core.space.spi.SpaceTemplateService</target-component>
+            <component-plugin>
+              <name>Space Template Configuration</name>
+              <set-method>registerSpaceTemplatePlugin</set-method>
+              <type>org.exoplatform.social.core.space.SpaceTemplateConfigPlugin</type>
+              <init-params>
+                <object-param>
+                  <name>template</name>
+                  <description>Space Template</description>
+                  <object type="org.exoplatform.social.core.space.SpaceTemplate">
+                    <field name="name"><string>custom</string></field>
+                    <field name="visibility"><string>private</string></field>
+                    <field name="registration"><string>open</string></field>
+                    <field name="bannerPath"><string>war:/conf/social-extension/social/space-template/custom/banner.png</string></field>
+                    <field name="homePageApplication">
+                      <object type="org.exoplatform.social.core.space.SpaceApplication">
+                        <field name="portletApp"><string>social-portlet</string></field>
+                        <field name="portletName"><string>SpaceActivityStreamPortlet</string></field>
+                        <field name="appTitle"><string>Home</string></field>
+                      </object>
+                    </field>
+                    <!-- Default applications to be installed when creating a new space -->
```

Comment:
No need to add comment here, else, you will have to add comment everywhere

Reply:
ok